docs/testing/unit-tests.md

```diff
@@ -1,3 +1,247 @@
+### Snapshot testing
+See [Snapshot testing](snapshot-testing.md)
+
+### Examples
+
+#### Mocking HTTP requests with Nock
+
+[Nock](https://github.com/node-nock/nock) allows us to override HTTP requests, and unit test code in isolation. For example, you might have an action  that returns a GET request in a thunk:
+
+```javascript
+// action.js
+export function requestSomeData() {
+	return dispatch => {
+		return wpcom
+            .getSomeData()
+            .then( data => {
+				dispatch( {
+					type: SOME_DATA_SUCCESS,
+					data,
+				} );
+			} )
+			.catch( error => {
+				dispatch( {
+					type: SOME_DATA_FAILURE,
+					error,
+				} );
+			} );
+	};
+}
+```
+
+To mock this response, we can specify the nock parameters in a describe block and use `persist()` to intercept all requests in that block.
+
+Here's a simple example:
+
+```javascript
+import { requestSomeData } from '../actions';
+import useNock from 'test/helpers/use-nock';
+
+describe( 'requestSomeData', () => {
+	describe( 'successful response', () => {
+		useNock( nock => {
+			nock( 'https://public-api.wordpress.com:443' )
+				.persist()
+				.get( '/rest/v1.1/some-endpoint' )
+				.reply( 200, { hoo: 'ray!' } );
+		} );
+
+		test( 'dispatches success action object when successful request is completed', () => {
+			const mockDispatchCallback = jest.fn();
+            // Trigger the action to return the thunk
+			requestSomeData()( mockDispatchCallback ).then( () => {
+                expect( mockDispatchCallback.mock.calls[ 0 ][ 0 ] ).toEqual( {
+                    type: SOME_DATA_SUCCESS,
+                    data: { hoo: 'ray!' }
+                } );
+            } );
+		} );
+     } );
+} );
+```
+
+### Mocking dependencies
+
+#### Dependency injection
+
+Passing dependencies to a function as arguments can often make your code simpler to test. As a basic example, we have a utility function that checks whether a value exists in a list:
+
+One way to do it would be to reference a dependency in a higher scope:
+
+```javascript
+import VALID_VALUES_LIST from './constants'
+
+function isValueValid( value ) {
+	return VALID_VALUES_LIST.includes( value );
+}
+
+```
+
+Our happy path test would therefore have to import and use a value from `VALID_VALUES_LIST` in order to pass:
+
+`expect( isValueValid( VALID_VALUES_LIST[ 0 ] ) ).toBe( true );`
+
+
+The above assertion is testing two behaviours: 1) that the function can detect an item in a list, and 2) that it can detect an item in `VALID_VALUES_LIST`.
+
+But what if we don't care what's stored in `VALID_VALUES_LIST`, or if the list is fetched via an HTTP request, and we only want to test whether `isValueValid` can detect an item in a list?
+
+Let's refactor the original function so that it accepts the target list as an argument:
+
+```javascript
+function isValueValid( value, validValuesList = [] ) {
+	return validValuesList.includes( value );
+}
+```
+
+Now we can pass mock lists, and, as a bonus, test a few more scenarios:
+
```

Comment:
This is a nice example how to structure code to be easier to test. 👍 

I'm wondering if we could format it with `Not so good` and `Good` comments like above to make it easier to read.